lib/CL/pocl_build.c

```diff
@@ -202,7 +202,11 @@ process_options (const char *options, char *modded_options, char *link_options,
 
   size_t i = 1; /* terminating char */
   size_t needed = 0;
-  char *temp_options = strdup (options);
+  char *temp_options = (char*) malloc (strlen (options) + strlen ("-g ") + 1);
+  strcpy (temp_options, "");
```

Comment:
You could just calloc the memory, there's no need for strcpy'ing an empty string.